pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1614,25 +1614,31 @@ func buildTLS(refs AllowedReferences, tls *k8s.GatewayTLSConfig, namespace strin
 			out.Mode = istio.ServerTLSSettings_MUTUAL
 		}
 		if len(tls.CertificateRefs) != 1 {
-			// This is required in the API, should be rejected in validation
-			return nil, &ConfigError{Reason: InvalidTLS, Message: "exactly 1 certificateRefs should be present for TLS termination"}
-		}
-		cred, err := buildSecretReference(tls.CertificateRefs[0], namespace)
-		if err != nil {
-			return nil, err
-		}
-		credNs := defaultIfNil((*string)(tls.CertificateRefs[0].Namespace), namespace)
-		sameNamespace := credNs == namespace
-		if !sameNamespace && !refs.SecretAllowed(credentials.ToResourceName(cred), namespace) {
-			return nil, &ConfigError{
-				Reason: InvalidListenerRefNotPermitted,
-				Message: fmt.Sprintf(
-					"certificateRef %v/%v not accessible to a Gateway in namespace %q (missing a ReferenceGrant?)",
-					tls.CertificateRefs[0].Name, credNs, namespace,
-				),
+			if out.Mode == istio.ServerTLSSettings_MUTUAL && tls.CertificateRefs == nil {
+				// No certs -> use gateway's certs (i.e., ISTIO_MUTUAL mode)
+				out.Mode = istio.ServerTLSSettings_ISTIO_MUTUAL
```

Comment:
I don't think we should have this implicit - MUTUAL and ISTIO_MUTUAL are quite different.

Also, Per L1622, will this be rejected by CRD validation anyways?

Reply:
Whether we make ISTIO_MUTUAL implicit or an explicit option, we will want 0 certificateRefs in the ISTIO_MUTUAL case, so won't it fail CRD validation either way? Is there going to be an official validator that needs to pass as part of conformance testing? If so, won't it need to allow for some of these kind of implementation specific extensions?

Reply:
Its in CRD validation, so its impossible to even create an object like this. If we want 0 cert references we need to make an upstream API change.

Alternatively we could do something like `certificateRef: builtinSpiffeIdentity` somehow - but not sure how that would look in practice, feels a bit odd.

Reply:
Hmm, sounds like we (and maybe other implementations) will probably need an upstream change to allow this for GAMMA?

I'll close this and put it on the back-burner for now.